app/jsx/components/ScrollToTopComp.jsx

```diff
@@ -5,7 +5,10 @@ import { withRouter } from 'react-router'
 
 class ScrollToTopComp extends React.Component {
   componentDidUpdate(prevProps) {
-    if (this.props.location.pathname !== prevProps.location.pathname) {
+    let this_search = this.props.location.search
+    let prev_search = prevProps.location.search
+    if (this.props.location.pathname !== prevProps.location.pathname ||
+      (this_search.includes("start") && !this_search.includes("start=0") && this_search !== prev_search)) {
```

Comment:
This is now working pretty well for me. Good job.

I'll note that it's slightly annoying that it scrolls to the very top of the search page, dominated by Informational Pages, when I'm stepping through Scholarly Works results, but that'd be much harder to fix.